sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosCommonRequestOptions.java

```diff
@@ -0,0 +1,311 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.ConsistencyLevel;
+import com.azure.cosmos.CosmosDiagnosticsThresholds;
+import com.azure.cosmos.CosmosEndToEndOperationLatencyPolicyConfig;
+import com.azure.cosmos.implementation.apachecommons.collections.list.UnmodifiableList;
+
+import java.util.List;
+
+/**
+ * The common request options for operations. This class should be used with the addPolicy method in the {@link com.azure.cosmos.CosmosClientBuilder}
+ *  to change request options without restarting the application.
+ *
+ */
+public final class CosmosCommonRequestOptions implements ReadOnlyCommonRequestOptions {
```

Comment:
Any reason why we need the Common - why not just CosmosRequestOptions (and then based on decision above CosmsoOperationSettings etc. or keeping CosmosRequestOptions)

Reply:
Changed to CosmosRequestOptions. I think that changing it to operation settings would be confusing because we already have so many request options. For the CosmosOperationDetails and CosmosOperationPolicy, I think it is fine to change to using operation naming as these are completely new.